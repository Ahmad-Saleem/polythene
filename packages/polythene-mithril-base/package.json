{
  "name": "polythene-mithril-base",
  "version": "1.0.0-rc.6",
  "description": "",
  "main": "dist/polythene-mithril-base.js",
  "module": "dist/polythene-mithril-base.mjs",
  "scripts": {
    "clean": "rm -rf dist && mkdir dist",
    "rollup": "../../node_modules/rollup/bin/rollup -c ../../scripts/rollup.umd.js && ../../node_modules/rollup/bin/rollup -c ../../scripts/rollup.es.js",
    "build": "yarn run rollup"
  },
  "files": [
    "dist",
    "README.md"
  ],
  "dependencies": {
    "mithril": "1.1.3",
    "polythene-core": "^1.0.0-rc.6"
  },
  "author": "Arthur Clemens <arthurclemens@gmail.com> (http://visiblearea.com)",
  "license": "MIT"
}
