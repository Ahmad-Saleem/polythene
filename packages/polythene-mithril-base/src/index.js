export { keys } from "./keys";
export { renderer } from "./renderer";
export { StateComponent } from "./StateComponent";
export { ViewComponent } from "./ViewComponent";
export { Toggle } from "./Toggle";
