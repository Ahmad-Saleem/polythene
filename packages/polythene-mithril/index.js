export * from "polythene-mithril-base";
export * from "polythene-mithril-button";
export * from "polythene-mithril-fab";
export * from "polythene-mithril-icon";
export * from "polythene-mithril-raised-button";
export * from "polythene-mithril-ripple";
export * from "polythene-mithril-shadow";
export * from "polythene-mithril-svg";
