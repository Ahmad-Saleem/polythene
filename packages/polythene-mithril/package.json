{
  "name": "polythene-mithril",
  "version": "1.0.0-rc.1",
  "description": "",
  "main": "dist/polythene-mithril.js",
  "module": "dist/polythene-mithril.mjs",
  "scripts": {
    "clean": "rm -rf dist && mkdir dist",
    "rollup": "../../node_modules/rollup/bin/rollup -c ../../scripts/rollup.umd.js && ../../node_modules/rollup/bin/rollup -c ../../scripts/rollup.es.js",
    "build": "npm run rollup"
  },
  "files": [
    "dist",
    "README.md"
  ],
  "dependencies": {
    "polythene-mithril-base": "1.0.0-rc.1",
    "polythene-mithril-button": "1.0.0-rc.1",
    "polythene-mithril-card": "1.0.0-rc.1",
    "polythene-mithril-checkbox": "1.0.0-rc.1",
    "polythene-mithril-dialog": "1.0.0-rc.1",
    "polythene-mithril-dialog-pane": "1.0.0-rc.1",
    "polythene-mithril-fab": "1.0.0-rc.1",
    "polythene-mithril-icon": "1.0.0-rc.1",
    "polythene-mithril-icon-button": "1.0.0-rc.1",
    "polythene-mithril-list": "1.0.0-rc.1",
    "polythene-mithril-list-tile": "1.0.0-rc.1",
    "polythene-mithril-menu": "1.0.0-rc.1",
    "polythene-mithril-radio-button": "1.0.0-rc.1",
    "polythene-mithril-raised-button": "1.0.0-rc.1",
    "polythene-mithril-ripple": "1.0.0-rc.1",
    "polythene-mithril-shadow": "1.0.0-rc.1",
    "polythene-mithril-svg": "1.0.0-rc.1",
    "polythene-mithril-switch": "1.0.0-rc.1",
    "polythene-mithril-tabs": "1.0.0-rc.1",
    "polythene-mithril-textfield": "1.0.0-rc.1",
    "polythene-mithril-toolbar": "1.0.0-rc.1"
  },
  "author": "Arthur Clemens <arthurclemens@gmail.com> (http://visiblearea.com)",
  "license": "MIT"
}

