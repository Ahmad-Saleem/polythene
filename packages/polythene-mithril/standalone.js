export * from "./index";
import { addLayoutStyles, addFastClick } from "polythene-utilities";
import { addTypography, addRoboto } from "polythene-material-design";

addTypography();
addRoboto();
addLayoutStyles();
addFastClick();
