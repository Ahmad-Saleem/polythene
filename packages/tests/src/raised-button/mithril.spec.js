import { runSnapshots } from "../../scripts/mithril-snapshots";
import { renderer, raisedButton } from "polythene-mithril";
import { keys } from "polythene-mithril-base";
import specTests from "./spec-tests.js";
import mithrilTests from "./tests-mithril.js";

runSnapshots({
  tests: specTests({ raisedButton, renderer, keys }).concat(mithrilTests),
  renderer
});
