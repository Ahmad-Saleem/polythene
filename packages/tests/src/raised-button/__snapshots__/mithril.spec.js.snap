exports[`test Dark tone class + light theme class 1`] = `
"<div style=\"background: rgb(255, 255, 255);\" class=\"pe-light-tone\">
  <a class=\"pe-button pe-text-button pe-raised-button\" tabindex=
  \"0\">
  <div class=\"pe-button__content\">
    <div class=\"pe-shadow pe-shadow--animated\">
      <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
      <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
    </div>
    <div class=\"pe-ripple pe-ripple--constrained\">
      <div class=\"pe-ripple__mask\">
        <div class=\"pe-ripple__waves\"></div>
      </div>
    </div>
    <div class=\"pe-button__wash\"></div>
    <div class=\"pe-button__focus\"></div>
    <div class=\"pe-button__label\">
      Normal
    </div>
  </div></a><a class=
  \"pe-button pe-text-button pe-raised-button pe-button--disabled\"
  tabindex=\"-1\" disabled>
  <div class=\"pe-button__content\">
    <div class=\"pe-button__label\">
      Disabled
    </div>
  </div></a><a class=
  \"pe-button pe-text-button pe-raised-button tests-raised-button-themed-button\"
  tabindex=\"0\">
  <div class=\"pe-button__content\">
    <div class=\"pe-shadow pe-shadow--animated\">
      <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
      <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
    </div>
    <div class=\"pe-ripple pe-ripple--constrained\">
      <div class=\"pe-ripple__mask\">
        <div class=\"pe-ripple__waves\"></div>
      </div>
    </div>
    <div class=\"pe-button__wash\"></div>
    <div class=\"pe-button__focus\"></div>
    <div class=\"pe-button__label\">
      Theme
    </div>
  </div></a>
</div>
"
`;

exports[`test Dark tone class + light tone 1`] = `
"<div style=\"background: rgb(255, 255, 255);\">
  <a class=
  \"pe-button pe-text-button pe-raised-button pe-light-tone\"
  tabindex=\"0\">
  <div class=\"pe-button__content\">
    <div class=\"pe-shadow pe-shadow--animated\">
      <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
      <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
    </div>
    <div class=\"pe-ripple pe-ripple--constrained\">
      <div class=\"pe-ripple__mask\">
        <div class=\"pe-ripple__waves\"></div>
      </div>
    </div>
    <div class=\"pe-button__wash\"></div>
    <div class=\"pe-button__focus\"></div>
    <div class=\"pe-button__label\">
      Normal
    </div>
  </div></a><a class=
  \"pe-button pe-text-button pe-raised-button pe-button--disabled pe-light-tone\"
  tabindex=\"-1\" disabled>
  <div class=\"pe-button__content\">
    <div class=\"pe-button__label\">
      Disabled
    </div>
  </div></a><a class=
  \"pe-button pe-text-button pe-raised-button pe-light-tone tests-raised-button-themed-button\"
  tabindex=\"0\">
  <div class=\"pe-button__content\">
    <div class=\"pe-shadow pe-shadow--animated\">
      <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
      <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
    </div>
    <div class=\"pe-ripple pe-ripple--constrained\">
      <div class=\"pe-ripple__mask\">
        <div class=\"pe-ripple__waves\"></div>
      </div>
    </div>
    <div class=\"pe-button__wash\"></div>
    <div class=\"pe-button__focus\"></div>
    <div class=\"pe-button__label\">
      Theme
    </div>
  </div></a>
</div>
"
`;

exports[`test No options 1`] = `
"<a class=\"pe-button pe-text-button pe-raised-button\" tabindex=\"0\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
</div></a>
"
`;

exports[`test Option: animateOnTap (false) 1`] = `
"<a class=\"pe-button pe-text-button pe-raised-button\" tabindex=\"0\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-button__label\">
    Don\'t animate shadow
  </div>
</div></a>
"
`;

exports[`test Option: children 1`] = `
"<a class=\"pe-button pe-text-button pe-raised-button\" tabindex=\"0\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div><span>Child</span>
</div></a>
"
`;

exports[`test Option: class 1`] = `
"<a class=\"pe-button pe-text-button pe-raised-button class-x\"
tabindex=\"0\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
</div></a>
"
`;

exports[`test Option: className 1`] = `
"<a class=\"pe-button pe-text-button pe-raised-button className-x\"
tabindex=\"0\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
</div></a>
"
`;

exports[`test Option: content 1`] = `
"<a class=\"pe-button pe-text-button pe-raised-button\" tabindex=\"0\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div><span>Content</span>
</div></a>
"
`;

exports[`test Option: disabled (false) 1`] = `
"<a class=\"pe-button pe-text-button pe-raised-button\" tabindex=\"0\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-button__label\">
    Not disabled
  </div>
</div></a>
"
`;

exports[`test Option: disabled (true) 1`] = `
"<a class=
\"pe-button pe-text-button pe-raised-button pe-button--disabled\"
tabindex=\"-1\" disabled>
<div class=\"pe-button__content\">
  <div class=\"pe-button__label\">
    Disabled
  </div>
</div></a>
"
`;

exports[`test Option: disabled -- dark theme class 1`] = `
"<a class=
\"pe-button pe-text-button pe-raised-button pe-button--disabled\"
tabindex=\"-1\" disabled>
<div class=\"pe-button__content\">
  <div class=\"pe-button__label\">
    Label
  </div>
</div></a>
"
`;

exports[`test Option: element 1`] = `
"<button parentclassname=\"pe-button pe-text-button pe-raised-button\"
shadowcomponent=\"[object Object]\" element=\"button\"></button>
"
`;

exports[`test Option: id 1`] = `
"<a id=\"id-x\" class=\"pe-button pe-text-button pe-raised-button\"
tabindex=\"0\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
</div></a>
"
`;

exports[`test Option: inactivate (2) 1`] = `
"<a class=\"pe-button pe-text-button pe-raised-button\" tabindex=\"0\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-button__label\">
    Inactivated for 2s
  </div>
</div></a>
"
`;

exports[`test Option: inactive (false) 1`] = `
"<a class=\"pe-button pe-text-button pe-raised-button\" tabindex=\"0\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-button__label\">
    Not inactive
  </div>
</div></a>
"
`;

exports[`test Option: inactive (true) 1`] = `
"<a class=
\"pe-button pe-text-button pe-raised-button pe-button--inactive\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-button__label\">
    Inactive
  </div>
</div></a>
"
`;

exports[`test Option: ink (false) 1`] = `
"<a class=\"pe-button pe-text-button pe-raised-button\" tabindex=\"0\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-button__label\">
    No ink
  </div>
</div></a>
"
`;

exports[`test Option: label -- dark theme class (should be app's primary color) 1`] = `
"<a class=\"pe-button pe-text-button pe-raised-button\" tabindex=\"0\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-button__label\">
    Label
  </div>
</div></a>
"
`;

exports[`test Option: label 1`] = `
"<a class=\"pe-button pe-text-button pe-raised-button\" tabindex=\"0\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-button__label\">
    Label
  </div>
</div></a>
"
`;

exports[`test Option: raised (with option z: 2) 1`] = `
"<a class=\"pe-button pe-text-button pe-raised-button\" tabindex=\"0\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-2\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-2\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-button__label\">
    Raised to 2
  </div>
</div></a>
"
`;

exports[`test Option: raised (with option z: 5) 1`] = `
"<a class=\"pe-button pe-text-button pe-raised-button\" tabindex=\"0\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-5\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-5\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-button__label\">
    Raised to 5
  </div>
</div></a>
"
`;

exports[`test Option: selected 1`] = `
"<a class=
\"pe-button pe-text-button pe-raised-button pe-button--selected\"
tabindex=\"0\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-button__label\">
    Selected
  </div>
</div></a>
"
`;

exports[`test Option: style (colors) 1`] = `
"<a class=\"pe-button pe-text-button pe-raised-button\" tabindex=\"0\">
<div style=
\"background-color: rgb(239, 108, 0); color: rgb(255, 255, 255);\"
class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-button__label\">
    Styled
  </div>
</div></a>
"
`;

exports[`test Option: tabindex 1`] = `
"<a tabindex=\"3\" class=\"pe-button pe-text-button pe-raised-button\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
</div></a>
"
`;

exports[`test Option: url (with oncreate) 1`] = `
"<a class=\"pe-button pe-text-button pe-raised-button\" tabindex=\"0\"
href=\"#!/shadow\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-button__label\">
    Go to /#/shadow
  </div>
</div></a>
"
`;

exports[`test Option: wash (false) 1`] = `
"<a class=\"pe-button pe-text-button pe-raised-button\" tabindex=\"0\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-button__label\">
    No wash
  </div>
</div></a>
"
`;

exports[`test Themed button (should be red) 1`] = `
"<a class=
\"pe-button pe-text-button pe-raised-button tests-raised-button-themed-button\"
tabindex=\"0\">
<div class=\"pe-button__content\">
  <div class=\"pe-shadow pe-shadow--animated\">
    <div class=\"pe-shadow__bottom pe-shadow--z-1\"></div>
    <div class=\"pe-shadow__top pe-shadow--z-1\"></div>
  </div>
  <div class=\"pe-ripple pe-ripple--constrained\">
    <div class=\"pe-ripple__mask\">
      <div class=\"pe-ripple__waves\"></div>
    </div>
  </div>
  <div class=\"pe-button__wash\"></div>
  <div class=\"pe-button__focus\"></div>
  <div class=\"pe-button__label\">
    Themed button
  </div>
</div></a>
"
`;

exports[`test Themed button (with option disabled) 1`] = `
"<a class=
\"pe-button pe-text-button pe-raised-button pe-button--disabled tests-raised-button-themed-button\"
tabindex=\"-1\" disabled>
<div class=\"pe-button__content\">
  <div class=\"pe-button__label\">
    Disabled themed button
  </div>
</div></a>
"
`;
